syntax/lean3.vim

```diff
@@ -24,7 +24,41 @@ syn keyword leanCommand universe universes example axioms constants
 syn keyword leanCommand meta parameter parameters variable variables
 syn keyword leanCommand reserve precedence postfix prefix notation infix infixl infixr
 
-syn keyword leanKeyword begin by end
+syn keyword leanTactic abel abstract ac_mono ac_refl all_goals any_goals apply
+        \ apply_assumption apply_auto_param apply_congr apply_fun apply_instance
+        \ apply_opt_param apply_rules apply_with assoc_rewrite assume assumption
+        \ async by_cases by_contra by_contradiction cancel_denoms case cases
+        \ cases_matching casesm cases_type cc change choose classical clear
+        \ clear_aux_decl clear_except clear_value comp_val congr constructor
+        \ continuity contradiction contrapose conv to_lhs to_rhs
+        \ convert convert_to dec_trivial delta destruct done dsimp dunfold
+        \ eapply econstructor elide unelide equiv_rw equiv_rw_type erewrite
+        \ erw exact exacts exfalso existsi ext1 ext extract_goal fail_if_success
+        \ fapply fconstructor field_simp filter_upwards fin_cases finish clarify
+        \ safe focus from fsplit funext generalize generalize_hyp
+        \ generalize_proofs generalizes group guard_hyp guard_target h_generalize
+        \ have hint induction inhabit injection injections injections_and_clear
+        \ interval_cases intro intros introv itauto iterate left right let
+        \ library_search lift linarith linear_combination mapply match_target
+        \ measurability mono nlinarith noncomm_ring nontriviality norm_cast
+        \ norm_fin norm_num nth_rewrite nth_rewrite_lhs nth_rewrite_rhs
+        \ observe obtain omega pi_instance pretty_cases push_neg rcases refine
+        \ refine_struct refl reflexivity rename rename_var repeat replace
+        \ revert revert_after revert_deps revert_target_deps rewrite_search
+        \ ring ring2 ring_exp rintro rintros rotate rw rewrite rwa scc show
+        \ show_term simp simp_intros simp_result simp_rw simpa skip slice solve1
+        \ solve_by_elim sorry admit specialize split split_ifs squeeze_simp
+        \ squeeze_simpa squeeze_dsimp squeeze_scope subst subst_vars substs
+        \ subtype_instance success_if_fail suffices suggest swap swap_var
+        \ symmetry tautology tfae tidy trace trace_simp_set trace_state
+        \ transitivity transport triv trivial trunc_cases try type_check
+        \ unfold unfold1 unfold_cases unfold_coes unfold_projs unify_equations
+        \ use with_cases wlog zify
+        \ contained
+syn region leanTacticMode matchgroup=Label start='\<begin\>' end='\<end\>' contains=leanTactic
+syn match  leanTacticBy 'by' nextgroup=leanTactic
```

Comment:
This doesn't cover braces-`by` blocks but I couldn't think how to do so easily quite yet.

Reply:
Actually I guess an additional `syn match` with a region may do it now that I've thought again about it.